mypy/subtypes.py

```diff
@@ -209,6 +215,15 @@ def visit_partial_type(self, left: PartialType) -> bool:
         # This is indeterminate as we don't really know the complete type yet.
         raise RuntimeError
 
+    def visit_type_type(self, left: TypeType) -> bool:
+        right = self.right
+        if isinstance(right, TypeType):
+            return is_subtype(left.item, right.item)
+        if isinstance(right, CallableType):
+            return right.is_type_obj() and is_subtype(left.item, right.ret_type)
```

Comment:
Should we verify that the signature of `__init__` of the left type is compatible with the callable on the right?